Terminal.Gui/Application/Application.Mouse.cs

```diff
@@ -143,6 +143,12 @@ internal static void OnMouseEvent (MouseEvent mouseEvent)
 
         if (view is { })
         {
+#if DEBUG_IDISPOSABLE
+            if (view.WasDisposed)
+            {
+                throw new ObjectDisposedException (view.GetType ().FullName);
```

Comment:
I'm torn on this one...

GetType() is not something we should be using... But this is also a conditional only emitted if the symbol is defined, so.... I guess it's fine?

In any case, here's something for you, to make all your conditional throws much cleaner:

```csharp

    /// <summary>
    ///     Throws an <see cref="ObjectDisposedException"/> if the provided <paramref name="condition"/> is <see langword="true"/>.
    /// </summary>
    /// <typeparam name="T">
    ///     The type of the object that was disposed. Compiler will figure it out if you provide anything to
    ///     <paramref name="possiblyDisposedObject"/>.
    /// </typeparam>
    /// <param name="condition">The condition upon which, if true, the exception will be thrown. If false, method simply returns.</param>
    /// <param name="possiblyDisposedObject">The object that we are complaining about. Only used by the method for determining type.</param>
    /// <param name="name">The name of the object. If not specified, the full name of its type will be used.</param>
    /// <param name="message">A message to include with the exception. If not specified, "Object already disposed" is used.</param>
    /// <exception cref="ObjectDisposedException">If the condition is true.</exception>
    [Conditional ("DEBUG_IDISPOSABLE")]
    internal static void ThrowObjectDisposedExceptionIfTrue<T> (
        [DoesNotReturnIf (true)] this bool condition,
        T? possiblyDisposedObject = null,
        string? name = null,
        string? message = null
    ) where T : class, IDisposable
    {
        if (condition)
        {
            throw new ObjectDisposedException (name ?? typeof (T).FullName, message ?? "Object was already disposed.");
        }
    }
```

That is written as an extension method on bool but of course can also be called as a regular static method. 

If you don't want to stick it in a static class, just remove the `this` from the condition parameter.

That method replaces the entirety of this:

```csharp
#if DEBUG_IDISPOSABLE
            if (view.WasDisposed)
            {
                throw new ObjectDisposedException (view.GetType ().FullName);
            }
#endif
```

with JUST this:

```csharp
view.WasDisposed.ThrowObjectDisposedExceptionIfTrue(view);
```

or

```csharp
ThrowObjectDisposedExceptionIfTrue(view.WasDisposed,view);
```

No conditional required. The attribute took care of that.

Reply:
Also, the reason for all these disposal checks goes away if we properly handle lifetime and ownership of IDisposables.

I might move directly addressing some of that stuff up on my priority list. These situations represent pretty critical and potentially even dangerous (security-wise) bugs and should not make it into the final version.

Reply:
Oh another thing. If you like the extension method style but only want to touch  spcific file, you can define the class as:

```csharp
file static class NiftyThrowHelpers
{
  // Extension methods
}
```

The `file` access modifier makes the type only visible within that single file. You can even define classes with identical names multiple times in the same namespace, so long as they are in different files, if you do that. A potential use case for that is this sort of thing or platform-specific code with a unified interface.

It's even more private than private! 🤯

Reply:
Love this, but not for this PR. Please open a new issue for this concept.

Reply:
If not included here, I'll likely stick it in the driver one I'm working on, since I can make use of it there, too.

I'll just do it file-scoped.

Reply:
Did you already realize that when you click on a `Border` of an open `Menu` this also throw tis exception for the `Border` adornment? The reason is that the `Menu` was disposed and the `HandlePressed` method is grabbing the menu's `Border`.

Reply:
> Did you already realize that when you click on a `Border` of an open `Menu` this also throw tis exception for the `Border` adornment? The reason is that the `Menu` was disposed and the `HandlePressed` method is grabbing the menu's `Border`.

Suggestion on how to fix?

Reply:
> Suggestion on how to fix?

The ideal is the `View` class had a `Disposed` property that must also available in `Debug` and Release `modes`. Attempting accessing a disposed object, two situations may occur:
1 - The disposed object is already null and `FindDeepestView` may not see it and it's ok.
2 - The disposed object is not null and `FindDeepestView` should check for the `Disposed` property and not return it, continuing search for the view under that position.

Reply:
> Why not just use the `view` instance instead of using reflection by calling `GetType` on all occurrences?
> 
> `throw new ObjectDisposedException (nameof (view));`

GetType is banned by the AoT compiler.

Use a generic and `typeof(T)` or type checks when you would otherwise be trying to call GetType(). Don't pass Type objects around if you can help it. Even outside of native AoT, doing that breaks one of the biggest value propositions of the managed runtime and static analysis: type-safety.

The compiler can't tell ahead of time what a Type variable is going to be. But a type parameter is a compile-time construct that _does_ result in a closed model in the final JITed binary.